### PR TITLE
Alias outputs doc

### PR DESCRIPTION
**Pull request by anistark**
# Description of change

Added alias create, destroy docs.

## Links to any relevant issues

fixes issue #1111 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Documentation Enhancement

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work

### CODE REVIEW

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+<div className={'hide-code-block-extras'}>
+
+```typescript reference
+    https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/examples/how_tos/alias/create.ts#L53
+```
+
+</div>
+
+3. Retry transaction until included [`Transaction`](../../references/nodejs/classes/Account/#retrytransactionuntilincluded) that was sent.
```
Comment (Dr-Electron) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx:68]:
```suggestion
3. Retry transaction until included [`Transaction`](../../references/nodejs/classes/Account.md#retrytransactionuntilincluded) that was sent.
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+<div className={'hide-code-block-extras'}>
+
+```typescript reference
+    https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/examples/how_tos/alias/destroy.ts#L56
+```
+
+</div>
+
+3. Retry transaction until included [`Transaction`](../../references/nodejs/classes/Account/#retrytransactionuntilincluded) that was sent.
```
Comment (Dr-Electron) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx:66]:
```suggestion
3. Retry transaction until included [`Transaction`](../../references/nodejs/classes/Account.md#retrytransactionuntilincluded) that was sent.
```

**Review by Dr-Electron — CHANGES_REQUESTED** [review]
Two broken links

**Comment by Dr-Electron** [comment]
Btw those examples are not aligned and don't create the same output. You should make sure the same useful output is created in the example. You can compare for example how we did it in the native token examples

**Comment by Dr-Electron** [comment]
Also you don't import the correct lines of code 😅

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  - output
+  - nodejs
+  - python
+  - rust
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+The Alias Output is a specific implementation of a UTXO state machine. Alias ID, the unique identifier of an instance of the deployed state machine, is generated deterministically by the protocol and is not allowed to change in any future state transitions.
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx:18]:
```suggestion
An [Alias Output](https://wiki.iota.org/tips/tips/TIP-0018/#alias-output) is a specific implementation of a UTXO state machine. The Alias ID, the unique identifier of an instance of the deployed state machine, is generated deterministically by the protocol and is not allowed to change in future state transitions.
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+Alias Output represents an alias account in the ledger with two control levels and a permanentAlias Address. The account owns other outputs that are locked under Alias Address. The account keeps track of state transitions (`State Index` counter), controlled foundries (`Foundry Counter`) and anchors the layer 2 state as metadata into the UTXO ledger.
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx:20]:
```suggestion
An Alias Output represents an alias account in the ledger with two control levels and a permanent Alias Address. The account owns other outputs that are locked under the Alias Address. The account keeps track of state transitions (`State Index` counter) and controlled foundries (`Foundry Counter`) and anchors the layer 2 state as metadata into the UTXO ledger.
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+## Example Code
+
+<Tabs groupId="language" queryString>
+<TabItem value="rust" label="Rust">
+
+The following example will:
+
+1. Instantiate a [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/core/struct.Wallet.html), get Alice's `Account` which was
+   [created in the first guide](../accounts-and-addresses/create-account.mdx).
+2. Create an alias output transaction by calling the [`Account.create_alias_output()`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.CreateAliasParams.html) function.
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx:31]:
```suggestion
2. Create an Alias Output transaction by calling the [`Account.create_alias_output()`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.CreateAliasParams.html) function.
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+<div className={'hide-code-block-extras'}>
+
+```rust reference
+https://github.com/iotaledger/iota-sdk/blob/develop/sdk/examples/how_tos/alias/create.rs#L40
+```
+
+</div>
+
+3. Retry transaction until included [`Transaction`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.Account.html#method.retry_transaction_until_included) that was sent.
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx:41]:
```suggestion
3. Retry the [`Transaction`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.Account.html#method.retry_transaction_until_included) until included.
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+<div className={'hide-code-block-extras'}>
+
+```rust reference
+https://github.com/iotaledger/iota-sdk/blob/develop/sdk/examples/how_tos/alias/create.rs#L43
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx:46]:
```suggestion
https://github.com/iotaledger/iota-sdk/blob/develop/sdk/examples/how_tos/alias/create.rs#L43-L45
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+</div>
+
+</TabItem>
+<TabItem value="typescript-node" label="Typescript (Node.js)">
+
+The following example will:
+
+1. Instantiate a [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/core/struct.Wallet.html), get Alice's `Account` which was
+   [created in the first guide](../accounts-and-addresses/create-account.mdx) and [sync it](../accounts-and-addresses/check-balance.mdx).
+2. Create an alias output transaction by calling the [`Account.prepareCreateAliasOutput()`](../../references/nodejs/classes/Account.md#preparecreatealiasoutput) function.
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx:58]:
```suggestion
2. Create an Alias Output transaction by calling the [`Account.prepareCreateAliasOutput()`](../../references/nodejs/classes/Account.md#preparecreatealiasoutput) function, and then [send](../../references/nodejs/classes/Account.md#send) it.
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+<div className={'hide-code-block-extras'}>
+
+```typescript reference
+    https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/examples/how_tos/alias/create.ts#L53
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx:63]:
```suggestion
https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/examples/how_tos/alias/create.ts#L53-L55
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+```
+
+</div>
+
+3. Retry transaction until included [`Transaction`](../../references/nodejs/classes/Account.md#retrytransactionuntilincluded) that was sent.
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx:68]:
```suggestion
3. Retry  the [`Transaction`](../../references/nodejs/classes/Account.md#retrytransactionuntilincluded) until included.
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+<div className={'hide-code-block-extras'}>
+
+```typescript reference
+https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/examples/how_tos/alias/create.ts#L60
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx:73]:
```suggestion
https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/examples/how_tos/alias/create.ts#L60-L62
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+```
+
+</div>
+
+</TabItem>
+<TabItem value="python" label="Python">
+
+1. Instantiate a [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/core/struct.Wallet.html), get Alice's `Account` which was
+   [created in the first guide](../accounts-and-addresses/create-account.mdx) and [sync it](../accounts-and-addresses/check-balance.mdx).
+2. Create an alias output transaction by calling the [`Account.prepareCreateAliasOutput()`](../../references/python/iota_sdk/wallet/account.md#prepare_create_alias_output) function.
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/create.mdx:83]:
```suggestion
2. Prepare an Alias Output transaction by calling the [`Account.prepare_create_alias_output()`](../../references/python/iota_sdk/wallet/account.md#prepare_create_alias_output) function, and then [send](../../references/python/iota_sdk/wallet/account.md#send) it.
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+  - output
+  - nodejs
+  - python
+  - rust
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+You can destroy an alias output by ID. If the alias still owns any outputs when you try to destroy it, you will get an error.
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx:18]:
```suggestion
You can destroy an Alias Output using its ID. You will get an error if the alias still owns any outputs when you try to destroy it.
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+## Example Code
+
+<Tabs groupId="language" queryString>
+<TabItem value="rust" label="Rust">
+
+The following example will:
+
+1. Instantiate a [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/core/struct.Wallet.html), get Alice's `Account` which was
+   [created in the first guide](../accounts-and-addresses/create-account.mdx).
+2. Destroy an alias output transaction by calling the [`Account.burn()`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.Account.html#method.burn) function.
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx:29]:
```suggestion
2. Destroy an Alias Output transaction by calling the [`Account.burn()`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.Account.html#method.burn) function.
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+
+<div className={'hide-code-block-extras'}>
+
+```rust reference
+https://github.com/iotaledger/iota-sdk/blob/develop/sdk/examples/how_tos/alias/destroy.rs#L44
+```
+
+</div>
+
+3. Retry transaction until included [`Transaction`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.Account.html#method.retry_transaction_until_included) that was sent.
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx:39]:
```suggestion
3. Retry the [`Transaction`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/account/struct.Account.html#method.retry_transaction_until_included) until included.
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+
+<div className={'hide-code-block-extras'}>
+
+```rust reference
+https://github.com/iotaledger/iota-sdk/blob/develop/sdk/examples/how_tos/alias/destroy.rs#L47
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx:44]:
```suggestion
https://github.com/iotaledger/iota-sdk/blob/develop/sdk/examples/how_tos/alias/destroy.rs#L47-L49
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+The following example will:
+
+1. Instantiate a [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/core/struct.Wallet.html), get Alice's `Account` which was
+   [created in the first guide](../accounts-and-addresses/create-account.mdx) and [sync it](../accounts-and-addresses/check-balance.mdx).
+2. Destroy an alias output transaction by calling the [`Account.preparedestroyalias()`](../../references/nodejs/classes/Account.md#preparedestroyalias) function.
+
+<div className={'hide-code-block-extras'}>
+
+```typescript reference
+    https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/examples/how_tos/alias/destroy.ts#L56
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx:61]:
```suggestion
https://github.com/iotaledger/iota-sdk/blob/develop/bindings/nodejs/examples/how_tos/alias/destroy.ts#L56-L58
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+</div>
+
+</TabItem>
+<TabItem value="typescript-node" label="Typescript (Node.js)">
+
+The following example will:
+
+1. Instantiate a [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/core/struct.Wallet.html), get Alice's `Account` which was
+   [created in the first guide](../accounts-and-addresses/create-account.mdx) and [sync it](../accounts-and-addresses/check-balance.mdx).
+2. Destroy an alias output transaction by calling the [`Account.preparedestroyalias()`](../../references/nodejs/classes/Account.md#preparedestroyalias) function.
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx:56]:
```suggestion
2. Prepare the transaction to destroy an Alias Output by calling the [`Account.preparedestroyalias()`](../../references/nodejs/classes/Account.md#preparedestroyalias) function, and then [send](../../references/nodejs/classes/Account.md#send) it.
```

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+```
+
+</div>
+
+</TabItem>
+<TabItem value="python" label="Python">
+
+1. Instantiate a [`Wallet`](https://docs.rs/iota-sdk/latest/iota_sdk/wallet/core/struct.Wallet.html), get Alice's `Account` which was
+   [created in the first guide](../accounts-and-addresses/create-account.mdx) and [sync it](../accounts-and-addresses/check-balance.mdx).
+2. Destroy an alias output transaction by calling the [`Account.prepare_destroy_alias()`](../../references/python/iota_sdk/wallet/account.md#prepare_destroy_alias) function.
```
Comment (lucas-tortora) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx:81]:
```suggestion
2. Destroy the transaction that will destroy the alias output by calling the [`Account.prepare_destroy_alias()`](../../references/python/iota_sdk/wallet/account.md#prepare_destroy_alias) function, and then [send](../../references/python/iota_sdk/wallet/account.md#send) it.
```

**Review by lucas-tortora — CHANGES_REQUESTED** [review]

**Comment by Dr-Electron** [comment]
> Btw those examples are not aligned and don't create the same output. You should make sure the same useful output is created in the example. You can compare for example how we did it in the native token examples

Are you going to work on it, or do you want to create another PR afterwards?

**Comment by anistark** [comment]
> > Btw those examples are not aligned and don't create the same output. You should make sure the same useful output is created in the example. You can compare for example how we did it in the native token examples
> 
> Are you going to work on it, or do you want to create another PR afterwards?

Python output is left to match. Will open a PR to sdk and then come back to this. Let it stay open till then?

**Comment by anistark** [comment]
The examples don't run on my system as usual. So, not able to test it out. nodejs and rust outputs are same now.
Here's my [PR](https://github.com/iotaledger/iota-sdk/pull/1212)

File: docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx (hunk 13 above)
Comment (luca-moser) [inline docs/build/iota-sdk/1.0.0/docs/how-tos/alias/destroy.mdx:18]: This is not correct, whether an alias output owns other outputs or not does matter when you are destroying one. There is nothing in place from a protocol level that would prevent you from burning/deleting an alias output of which its address are still present in other addresses. Likewise, you could also theoretically create outputs which are owned by an Alias address which does not even exist.

Can you maybe point out how you came to this conclusion?

**Review by luca-moser — CHANGES_REQUESTED** [review]

**Comment by Dr-Electron** [comment]
> The examples don't run on my system as usual. So, not able to test it out. nodejs and rust outputs are same now.
> Here's my [PR](https://github.com/iotaledger/iota-sdk/pull/1212). If you're free, please check out python code.

Unrelated question, but:
Did you create an issue to track this btw? I forgot if you did 😅

**Comment by Dr-Electron** [comment]
Can you please resolve the conflict. Let's get this merged 🦾

**Review by luca-moser — APPROVED** [review]

**Review by luca-moser — APPROVED** [review]

**Pull request merged**